### ISSUE TITLE
Wildboar ShapeletForestClassifier not working?

### ISSUE THREAD

**Issue by madarax64** [isaksamsten/wildboar#22]
Hello,
I was trying to use your package, so I did a `pip install wildboar` started out with the tutorial code as follows:

`from wildboar.datasets import load_synthetic_control, load_two_lead_ecg`
`x_train, x_test, y_train, y_test = load_two_lead_ecg(merge_train_test=False)`
`from wildboar.ensemble import ShapeletForestClassifier`
`clf = ShapeletForestClassifier()`
`clf.fit(x_train, y_train)`
`print(clf.predict(x_test[-1:, :]))`

But rather than give 6 as expected, it gives 2 as output. 
I tried many other UCR datasets, as I have used an older version of the ShapeletForestClassifier before to great effect, but the accuracy on all the ones I tried (FordA, Ham, Gun_Point, CBF,Cricket_X,...) are all horribly poor. Perhaps there is something broken in the release, as I know the ShapeletForestClassifier performs very well?

Thanks in advance

**Comment by isaksamsten** [comment]
Hi!

Thanks for your bug report!

The example in the tutorial seems to be a little unintuitive since I load two datasets (synthetic_control and two_lead_ecg) and define x_train etc two times. The actual output is from the not where the training and testing data is split from the full data (i.e., from synthetic_control) (which contains the classes 1-6, thus 6 is a possible outcome). In the code you submit as part of this issue the two_lead_ecg dataset is used which only contains the classes 1 and 2 so the output of your example actually make sense.

If i run your example the resulting model has an accuracy of 90% and if I change the metric to "scaled_euclidean" the accuracy the expected 96% accuracy.

As for the other examples the accuracy i get using the following snippet:

```
def test_dataset(dataset):
    x_train, x_test, y_train, y_test = load_dataset(dataset, merge_train_test=False)
    clf =ShapeletForestClassifier(metric="scaled_euclidean", random_state=123, n_jobs=-1)
    clf.fit(x_train, y_train)
    return clf.score(x_test, y_test)
```

These are the results I get:
```
>>> test_dataset("GunPoint")
0.9866666666666667
>>> test_dataset("CricketX")
0.764102564102564
>>> test_dataset("CBF")
0.9722222222222222
```

To fully resolve the issue could you tell me your:

* Operating system
* Wildboar version
* Python version

Thanks again your bug report!

**Comment by madarax64** [comment]
Hello,
Thanks for your response. I've attached a screenshot of what I get when I run your code above - tl;dr its still giving pretty poor accuracy (48%). 

![Snip](https://user-images.githubusercontent.com/4976654/118613728-f064c980-b7b6-11eb-85f2-b787e785026f.PNG)

My Operating system is Windows 10 20H2, Wildboar version is 1.0.9 and Python version is 3.7.10. Incidentally I'm working in a miniconda environment. However I tried this on my native/non-anaconda Python environment (also Python 3.7.9) and had the same problem.

**Comment by isaksamsten** [comment]
Thanks for your report!

I will try to debug the issue once I get hold of a windows computer.

Could you try with at 64bit installation of miniconda?

**Comment by madarax64** [comment]
Hello,
Thanks for your prompt response!
Sadly, no - I've only the one computer. But, the other Python installation I told you about (i.e the non-Anaconda one I also have) is 64-bit, and shows the same issue..

**Comment by isaksamsten** [comment]
The issue seems to affect only Windows. Pre-build binaries and code compiled from source.

**Comment by isaksamsten** [comment]
Fixed in version 1.0.10.

**Comment by madarax64** [comment]
Confirmed fixed, thanks! Got 1.0.10 from pip. Thanks again :)
